test/unittests/test_huffman.cpp

```diff
@@ -0,0 +1,152 @@
+#include "algorithms/synthesis/encoding.hpp"
+#include "core/io/pla_parser.hpp"
+
+#include "gtest/gtest.h"
+
+using namespace dd::literals;
+using namespace syrec;
+
+class TestHuff: public testing::Test {
+protected:
+    TruthTable  tt{};
+    std::string test_circuits_dir = "./circuits/";
+
+    TruthTable::Cube::Value emptyVal;
+
+    TruthTable::Cube c00{};
+    TruthTable::Cube c01{};
+    TruthTable::Cube c10{};
+    TruthTable::Cube c11{};
+
+    TruthTable::Cube c000{};
+    TruthTable::Cube c001{};
+    TruthTable::Cube c010{};
+    TruthTable::Cube c011{};
+
+    TruthTable::Cube c0_{};
+    TruthTable::Cube c0__{};
+    TruthTable::Cube c1__{};
+
+    void SetUp() override {
+        c00 = TruthTable::Cube::fromInteger(0b00U, 2U);
+        c01 = TruthTable::Cube::fromInteger(0b01U, 2U);
+        c10 = TruthTable::Cube::fromInteger(0b10U, 2U);
+        c11 = TruthTable::Cube::fromInteger(0b11U, 2U);
+
+        c000 = TruthTable::Cube::fromInteger(0b000U, 3U);
+        c001 = TruthTable::Cube::fromInteger(0b001U, 3U);
+        c010 = TruthTable::Cube::fromInteger(0b010U, 3U);
+        c011 = TruthTable::Cube::fromInteger(0b011U, 3U);
+
+        c0_.emplace_back(false);
+        c0_.emplace_back(emptyVal);
+
+        c0__.emplace_back(false);
+        c0__.emplace_back(emptyVal);
+        c0__.emplace_back(emptyVal);
+
+        c1__.emplace_back(true);
+        c1__.emplace_back(emptyVal);
+        c1__.emplace_back(emptyVal);
```

Comment:
Similar to the `Cube::fromInteger` method there should be a `Cube::fromString` method that can handle don't cares. Then constructions like these are considerably simplified.

Furthermore, it might make sense to not only provide the `equals()` and `find()` functions mentioned in the other comment for integers, but also for strings.

This allows to remove all these variable definitions here as well.

Reply:
done 👍🏼